feature : create the share class

Noor-Serry opened the pull request.

Noor-Serry committed:
feature : create the share class

Belal-Alkhatib commented on chat/src/main/kotlin/chat/dto/ContactResponse.kt:
```diff
@@ -0,0 +1,13 @@
+import java.util.UUID
+
+class ContactResponse(
+    val id: UUID = UUID.randomUUID(),
```
we don't need to reassign the id here this class is just for mapping the domain class to another format as an API response. Am I right?

Noor-Serry replied:
we need to send it to receive it agian when the user click on the contact 
that same when we was reciving the movie id and when the user click on the movie we was sending it to the backend to get its details .

omer1998 replied:
yes noor the problem is only you asigned it here as a random all the time, but this is not the case it will be as you said the uuid associated with this contact 
although it will be overrided but it seem a bit wierd if specify it as a random value here

Noor-Serry replied:
I got you

Belal-Alkhatib commented on chat/src/main/kotlin/chat/dto/ContactResponse.kt:
```diff
@@ -0,0 +1,13 @@
+    val phoneNumber: String,
+    val userId: UUID,
+    val imageUrl: String,
+    val isMENAMember: Boolean
```
rename `isMENAMember `to `isMenaMember `to follow Kotlin naming conventions 😅

Noor-Serry replied:
done

Noor-Serry committed:
Rename isMENAMember to isMenaMember in ContactResponse.kt

omer1998 reviewed (commented):
good job noor, what do you think about my comments

omer1998 commented on chat/src/main/kotlin/chat/entity/Contact.kt:
```diff
@@ -0,0 +1,22 @@
+    @Column(nullable = false) val name: String,
+    @Column(nullable = false) val phoneNumber: String,
+    @Column(nullable = false) val userId: UUID,
+    @Column(nullable = false) val imageUrl: String,
```
the image url will be the image of mena user am i right so do we need to save it again here in this table

Noor-Serry replied:
I think if the contact has an account in Mena, we will display their profile image. If not, we will display the image we received when the user sent it

Noor-Serry replied:
<img width="447" height="605" alt="image" src="https://github.com/user-attachments/assets/388a731b-54b2-4d2d-b238-ec34d53db856" />
here there are accounts that is not in mena and its have images

Belal-Alkhatib replied:
as the designer mentioned, we will only display the Mena profile image if available, and ignore any images coming from the user’s phone (no upload or save).

Noor-Serry committed:
Update ContactResponse id and userId to String

The `id` and `userId` fields in `ContactResponse` were changed from `UUID` to `String`. This allows for more flexibility in the types of identifiers that can be used.

Noor-Serry committed:
Make Contact.imageUrl nullable

The `imageUrl` property of the `Contact` entity is now nullable. This allows for contacts to be created or stored without requiring an image URL.

Noor-Serry committed:
Make Contact.imageUrl nullable

The `imageUrl` property of the `Contact` entity is now nullable. This allows for contacts to be created or stored without requiring an image URL.

Noor-Serry committed:
Merge remote-tracking branch 'origin/feature/chat-setup' into feature/chat-setup

# Conflicts:
#	chat/src/main/kotlin/chat/entity/Contact.kt

omer1998 commented on chat/src/main/kotlin/chat/entity/Contact.kt:
```diff
@@ -0,0 +1,28 @@
+    val user: User,
+    @Transient var imageUrl: String? = null,
+    @Transient var isMenaMember: Boolean = false
+)
```
i don't see the owner user of this contact 
the user relationship defined is concerned with the situation that this contact is related to another mena user in our system 
i am thinking in this way:
the each contact is owned by user and this contact might be another mena user in our system (relationship) am i right
another thing we need to make this user nullable because not every contact we add will be actually a mena user in our system 

also if we know the mena user related to this contact (not the owner) we don't need to put imagUrl or isMenaMember 
as we can know them dirctely

Belal-Alkhatib commented on chat/src/main/kotlin/chat/dto/ContactResponse.kt:
```diff
@@ -0,0 +1,10 @@
+    val name: String,
+    val phoneNumber: String,
+    val userId: String,
+    val imageUrl: String,
```
in case there is a contact that is not a mena user then `imageUrl `should be `null`, or just we will make it **empty** ?

Can you tell me why you decided to add the `user id` as a parameter here in the response class ?

Joseph-Sameh-0 replied:
@Belal-Alkhatib I think we will use userId for starting chat but it's not now

and yes the imageUrl should ba nullable

Belal-Alkhatib reviewed (approved):
excellent effort @Noor-Serry 👏

Joseph-Sameh-0 commented on chat/src/main/kotlin/chat/entity/Contact.kt:
```diff
@@ -0,0 +1,28 @@
+@Entity
+@Table(name = "contacts")
+data class Contact(
+    @Id @Column(
+        columnDefinition = "uuid",
+        updatable = false,
+        nullable = false
+    ) val id: UUID = UUID.randomUUID(),
```
did you discuss about the id will be string or integer?

Noor-Serry committed:
Refactor ContactResponse and Contact entities

- Convert ContactResponse to a data class.
- Remove `userId` from `ContactResponse`.
- Make `imageUrl` nullable in `ContactResponse`.
- Rename `user` to `ownerUser` in `Contact` entity.
- Remove `imageUrl` and `isMenaMember` transient properties from `Contact` entity.
- Enable `TYPESAFE_PROJECT_ACCESSORS` feature preview in `settings.gradle.kts`.
- Add `identity` project implementation dependency to `chat/build.gradle.kts`.
